### PR TITLE
Build: update to Node 6.9.0

### PR DESCRIPTION
**Pull request by gwwar**
This PR updates Calypso to node v6.9.0 [Changelog](https://nodejs.org/en/blog/release/v6.9.0/) 

 v6.9.0 marks the transition of Node.js v6 into Long Term Support (LTS) with the codename "Boron" and also includes minor security fixes.

### Local Dev
- Switch to node version 6.9.0 (npm 3.10.8)
- `make distclean`
- `make run`
- No functional changes to Calypso

### Local Docker Image
- Follow image instructions at: PCYsg-5XR-p2

### Desktop
- Clone the wp-desktop project
- Follow install instructions
- In the calypso submodule checkout this branch
- Desktop behaves normally

cc @mtias @blowery @lamosty @gziolo @rralian @aduth  @sendhil @dmsnell

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -117,6 +116,7 @@ jsLoader = {
 };
 
 if ( CALYPSO_ENV === 'development' ) {
+	const DashboardPlugin = require( 'webpack-dashboard/plugin' );
```
Comment (gwwar) [inline webpack.config.js]: screen shot 2016-10-18 at 2 06 22 pm

@dmsnell looks like I'm running into some issues with the webpack dashboard-plugin in the docker build. Looks like we'll need to move that require.

I think npm has updated it's install behavior with regards to the shrinkwrap. I now see differences between `npm install` vs `npm install --production` with a shrinkwrap that has dev dependencies.

**Comment by gwwar** [comment]
screen shot 2016-10-18 at 3 15 12 pm

running the dockerfile with  `npm install --production` runs into this issue.

It appears to clear up if you use `npm install`
```
bash-3.2$ npm ls tapable
wp-calypso@0.17.0 /Users/kerryliu/checkouts/wp-calypso
└─┬ webpack@1.13.1
  └── tapable@0.1.10 
```

**Comment by sendhil** [comment]
Works for me on Desktop.

**Comment by gwwar** [comment]
Hmm, maybe I need to up my memory locally? Could someone else test on docker with a fresh image?

**Comment by blowery** [comment]
@gwwar weird, working for me too, in docker. hrm.

**Comment by gwwar** [comment]
Oh sorry forgot to mention that I also built it from a clean base. (No previous images)  `docker rmi $(docker images -q)`

**Comment by gwwar** [comment]
Ok I can't repo the error anymore after a reboot + docker update ¯\_(ツ)_/¯

**Comment by dmsnell** [comment]
> (No previous images) docker rmi $(docker images -q)

whoa! killing them all! this is a good reason to use the `--name` parameter so we can kill only the Calypso image (we're not the only ones in the universe)

```bash
docker rmi wp-calypso
```

**Comment by gwwar** [comment]
> whoa! killing them all!

Oh sorry, was clearing out a bunch of old images 😄

**Comment by gwwar** [comment]
👍 Alright the tested docker image looks OK in an wp-calypso env. Unless folks find any issues I'll aim to land this on Monday

**Comment by dmsnell** [comment]
Tested fine for me @gwwar!

:shipit:

**Comment by gziolo** [comment]
https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V6.md#6.9.1. There is another version which fixes some regression from 6.8.0. It was release only yesterday ;)

**Comment by gwwar** [comment]
Thanks for the heads up @gziolo

**Pull request merged**